url89cafed56a_AlexDiru_esper_compiler_final_tgz-pJ8-compiler_ProgramJ8/src/compiler/EsperCGenerator.java

```diff
@@ -0,0 +1,519 @@
+package compiler;
+
+import java.util.ArrayList;
+
+public class EsperCGenerator {
+
+	private static ArrayList<VariableInformation> variableList;
+
+	private static String getVariableTypeFromName(String name) {
+
+		for (VariableInformation var : variableList)
+			if (var.name.equals(name))
+				return var.type;
+
+		if (name.equals("inf") || name.equals("-inf") || name.equals("nullity"))
+			return "transreal";
+
+		return "unknown";
+	}
+
+	/** Generates the left brace with the required indentation
+	 * @param indent The number of tabs to indent with
+	 * @return The string with the left brace */
+	private static String leftBrace(int indent) {
+		return "\n" + getIndentString(indent) + "{\n";
+	}
+
+	/** Generates the right brace with the required indentation
+	 * @param indent The number of tabs to indent with
+	 * @return The string with the right brace */
+	private static String rightBrace(int indent) {
+		return getIndentString(indent) + "}\n";
+	}
+
+	/** Gets the string with the correct number of indentations
+	 * @param indent The number of indentations
+	 * @return The string of indentation */
+	private static String getIndentString(int indent) {
+		String indentString = "";
+
+		for (int i = 0; i < indent; i++)
+			indentString += "\t";
+		return indentString;
+	}
+
+	/** Generates the C code for the given tree
+	 * @param parseRoot The root node of the tree to parse
+	 * @param tvariableList The list of variables in the code
+	 * @return The generated C code */
+	public static String generate(ParseTree parseRoot, ArrayList<VariableInformation> tvariableList) {
+
+		variableList = tvariableList;
+
+		String code = "#include <stdio.h>\n\n";
+		code += getTransrealLibrary();
+		code += "int main() \n{\n";
+
+		for (int i = 0; i < parseRoot.children.size(); i++) {
+			code += generateNode(parseRoot.children.get(i), 1);
+		}
+
+		return code + "}\n";
+	}
+
+	/** Generates C code for the given node
+	 * @param parseRoot The node to generate the code for
+	 * @param indent The number of indents
+	 * @return The C code */
+	private static String generateNode(ParseTree parseRoot, int indent) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 33. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.